File: app/bmi/src/main/java/com/example/bmi/MenuActivity.java
package com.example.bmi;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

/**
 * Application main menu
 *
 * @author Rafał Sochacki s20047
 */
public class MenuActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        Button goBmiBtn, goCaloriesBtn, goReceip, goQuiz;

        goBmiBtn = findViewById(R.id.bmiCalculatorButton);
        goCaloriesBtn = findViewById(R.id.caloriesCalculator);
        goReceip = findViewById(R.id.getReceip);
        goQuiz = findViewById(R.id.goQuiz);

        goBmiBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                launchBmiCalculator();
            }
        });

        goCaloriesBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                launchCaloriesCalculator();
            }
        });

        goReceip.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                launchGetReceip();
            }
        });

        goQuiz.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                launchQuiz();
            }
        });
    }

    private void launchBmiCalculator() {
        Intent intent = new Intent(this, BmiCalculatorActivity.class);
        startActivity(intent);
    }

    private void launchCaloriesCalculator() {
        Intent intent = new Intent(this, CaloriesCalculatorActivity.class);
        startActivity(intent);
    }

    private void launchGetReceip() {
        Intent intent = new Intent(this, ReceipActivity.class);
        startActivity(intent);
    }

    private void launchQuiz() {
        Intent intent = new Intent(this, QuizActivity.class);
        startActivity(intent);
    }
}

File: app/bmi/src/main/java/com/example/bmi/MenuActivity.kt
package com.example.bmi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import android.widget.*

/**
 * Application main menu
 *
 * @author Rafał Sochacki s20047
 */
class MenuActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)
        val goBmiBtn: Button
        val goCaloriesBtn: Button
        val goReceip: Button
        val goQuiz: Button
        goBmiBtn = findViewById(R.id.bmiCalculatorButton)
        goCaloriesBtn = findViewById(R.id.caloriesCalculator)
        goReceip = findViewById(R.id.getReceip)
        goQuiz = findViewById(R.id.goQuiz)
        goBmiBtn.setOnClickListener { launchBmiCalculator() }
        goCaloriesBtn.setOnClickListener { launchCaloriesCalculator() }
        goReceip.setOnClickListener { launchGetReceip() }
        goQuiz.setOnClickListener { launchQuiz() }
    }

    private fun launchBmiCalculator() {
        val intent = Intent(this, BmiCalculatorActivity::class.java)
        startActivity(intent)
    }

    private fun launchCaloriesCalculator() {
        val intent = Intent(this, CaloriesCalculatorActivity::class.java)
        startActivity(intent)
    }

    private fun launchGetReceip() {
        val intent = Intent(this, ReceipActivity::class.java)
        startActivity(intent)
    }

    private fun launchQuiz() {
        val intent = Intent(this, QuizActivity::class.java)
        startActivity(intent)
    }
}